### DISCUSSION TITLE
[CONTRACTS] - Yield Farm + pool2 in uAR instead of UBQ

### DISCUSSION THREAD

**Discussion by 0x4007** [pavlovcik/uad-ui#143] [Ideas]
In case we must fallback to the yield farm idea, in order to preserve the `UBQ`'s value for the long term what if...

1. We allow `UBQ` holders to convert 1:1 `UBQ` -> `uAR`, burning their `UBQ`.
- This will allow those who want to exit to do so, while rewarding long term holders of `UBQ` by reducing the supply.
2. Incentivize `???-ETH` pairs with high APR in `uAR` (mostly other projects related in our sector of DeFi such as, but not limited to `LUSD`, `OHM`, `MIM`, `UST`, `FRAX`, `FEI`, `DOLA`, `DAI`, `USDC`, `USDT`)
3. `uAR-ETH` pool2 with 100x the rewards or something crazy, cause pool2.

This way, if people are left holding uAR they aren't dead in the water whenever we can get the TWAP to go above 1.00.

**Comment by 0x4007** [comment]
Found the original Sushiswap farm code representing the farms for reference.

```typescript
export const supportedPools = [
    {
      pid: 12,
      lpAddresses: {
        1: '0xce84867c3c02b05dc570d0135103d3fb9cc19433',
      },
      tokenAddresses: {
        1: '0x6b3595068778dd592e39a122f4f5a5cf09c90fe2',
      },
      name: 'Sushi Party!',
      symbol: 'SUSHI-ETH UNI-V2 LP',
      tokenSymbol: 'SUSHI',
      icon: '🍣',
    },
    {
      pid: 0,
      lpAddresses: {
        1: '0x0d4a11d5eeaac28ec3f61d100daf4d40471f1852',
      },
      tokenAddresses: {
        1: '0xdac17f958d2ee523a2206206994597c13d831ec7',
      },
      name: 'Tether Turtle',
      symbol: 'USDT-ETH UNI-V2 LP',
      tokenSymbol: 'USDT',
      icon: '🐢',
    },
    {
      pid: 1,
      lpAddresses: {
        1: '0xb4e16d0168e52d35cacd2c6185b44281ec28c9dc',
      },
      tokenAddresses: {
        1: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      },
      name: 'Circle Snail',
      symbol: 'USDC-ETH UNI-V2 LP',
      tokenSymbol: 'USDC',
      icon: '🐌',
    },
    {
      pid: 2,
      lpAddresses: {
        1: '0xa478c2975ab1ea89e8196811f51a7b7ade33eb11',
      },
      tokenAddresses: {
        1: '0x6b175474e89094c44da98b954eedeac495271d0f',
      },
      name: 'Donald DAI',
      symbol: 'DAI-ETH UNI-V2 LP',
      tokenSymbol: 'DAI',
      icon: '🦆',
    },
    {
      pid: 3,
      lpAddresses: {
        1: '0xf80758ab42c3b07da84053fd88804bcb6baa4b5c',
      },
      tokenAddresses: {
        1: '0x57ab1ec28d129707052df4df418d58a2d46d5f51',
      },
      name: 'Spartan Dollar',
      symbol: 'SUSD-ETH UNI-V2 LP',
      tokenSymbol: 'SUSD',
      icon: '🦍',
    },
    {
      pid: 7,
      lpAddresses: {
        1: '0x88d97d199b9ed37c29d846d00d443de980832a22',
      },
      tokenAddresses: {
        1: '0x04fa0d235c4abf4bcf4787af4cf447de572ef828',
      },
      name: 'Umami Squid',
      symbol: 'UMA-ETH UNI-V2 LP',
      tokenSymbol: 'UMA',
      icon: '🦑',
    },
    {
      pid: 9,
      lpAddresses: {
        1: '0xf421c3f2e695c2d4c0765379ccace8ade4a480d9',
      },
      tokenAddresses: {
        1: '0xba11d00c5f74255f56a5e366f4f77f5a186d7f55',
      },
      name: 'Band-osaurus',
      symbol: 'BAND-ETH UNI-V2 LP',
      tokenSymbol: 'BAND',
      icon: '🦖',
    },
    {
      pid: 8,
      lpAddresses: {
        1: '0xa2107fa5b38d9bbd2c461d6edf11b11a50f6b974',
      },
      tokenAddresses: {
        1: '0x514910771af9ca656af840dff83e8264ecf986ca',
      },
      name: 'Toadie Marine',
      symbol: 'LINK-ETH UNI-V2 LP',
      tokenSymbol: 'LINK',
      icon: '🐸',
    },
    {
      pid: 10,
      lpAddresses: {
        1: '0xc5be99a02c6857f9eac67bbce58df5572498f40c',
      },
      tokenAddresses: {
        1: '0xd46ba6d942050d489dbd938a2c909a5d5039a161',
      },
      name: 'Ample Chicks',
      symbol: 'AMPL-ETH UNI-V2 LP',
      tokenSymbol: 'AMPL',
      icon: '🐥',
    },
    {
      pid: 4,
      lpAddresses: {
        1: '0xcffdded873554f362ac02f8fb1f02e5ada10516f',
      },
      tokenAddresses: {
        1: '0xc00e94cb662c3520282e6f5717214004a7f26888',
      },
      name: 'Compound Truffle',
      symbol: 'COMP-ETH UNI-V2 LP',
      tokenSymbol: 'COMP',
      icon: '🍄',
    },
    {
      pid: 5,
      lpAddresses: {
        1: '0xab3f9bf1d81ddb224a2014e98b238638824bcf20',
      },
      tokenAddresses: {
        1: '0x80fb784b7ed66730e8b1dbd9820afd29931aab03',
      },
      name: 'Aave Boar',
      symbol: 'LEND-ETH UNI-V2 LP',
      tokenSymbol: 'LEND',
      icon: '🐗',
    },
    {
      pid: 6,
      lpAddresses: {
        1: '0x43ae24960e5534731fc831386c07755a2dc33d47',
      },
      tokenAddresses: {
        1: '0xc011a73ee8576fb46f5e1c5751ca3b9fe0af2a6f',
      },
      name: 'Synthetic Snake',
      symbol: 'SNX-ETH UNI-V2 LP',
      tokenSymbol: 'SNX',
      icon: '🐍',
    },
    {
      pid: 11,
      lpAddresses: {
        1: '0x2fdbadf3c4d5a8666bc06645b8358ab803996e28',
      },
      tokenAddresses: {
        1: '0x0bc529c00c6401aef6d220be8c6ea1667f6ad93e',
      },
      name: 'YFI Whale',
      symbol: 'YFI-ETH UNI-V2 LP',
      tokenSymbol: 'YFI',
      icon: '🐋',
    },
    {
      pid: 13,
      lpAddresses: {
        1: '0x8bd1661da98ebdd3bd080f0be4e6d9be8ce9858c',
      },
      tokenAddresses: {
        1: '0x408e41876cccdc0f92210600ef50372656052a38',
      },
      name: 'REN Rhino',
      symbol: 'REN-ETH UNI-V2 LP',
      tokenSymbol: 'REN',
      icon: '🦏',
    },
    {
      pid: 14,
      lpAddresses: {
        1: '0xaad22f5543fcdaa694b68f94be177b561836ae57',
      },
      tokenAddresses: {
        1: '0x68A118Ef45063051Eac49c7e647CE5Ace48a68a5',
      },
      name: 'BASED Bull',
      symbol: 'BASE-sUSD UNI-V2 LP',
      tokenSymbol: 'BASED',
      icon: '🐂',
    },
    {
      pid: 15,
      lpAddresses: {
        1: '0xcc3d1ecef1f9fd25599dbea2755019dc09db3c54',
      },
      tokenAddresses: {
        1: '0x476c5E26a75bd202a9683ffD34359C0CC15be0fF',
      },
      name: 'SRM Shark',
      symbol: 'SRM-ETH UNI-V2 LP',
      tokenSymbol: 'SRM',
      icon: '🦈',
    },
    {
      pid: 16,
      lpAddresses: {
        1: '0xa5904961f61bae7c4dd8478077556c91bf291cfd',
      },
      tokenAddresses: {
        1: '0xaba8cac6866b83ae4eec97dd07ed254282f6ad8a',
      },
      name: 'SUSHIv2 SUSHI',
      symbol: 'SUSHIv2-ETH UNI-V2 LP',
      tokenSymbol: 'SUSHIv2',
      icon: '🍠',
    },
    {
      pid: 17,
      lpAddresses: {
        1: '0x3da1313ae46132a397d90d95b1424a9a7e3e0fce',
      },
      tokenAddresses: {
        1: '0xD533a949740bb3306d119CC777fa900bA034cd52',
      },
      name: 'CRV Crocodile',
      symbol: 'CRV-ETH UNI-V2 LP',
      tokenSymbol: 'CRV',
      icon: '🐊',
    },
  ]
```

**Comment by 0x4007** [comment]
### pool2
`uAR-ETH`
### pool1

`uAD-ETH`
`UBQ-ETH`
`LUSD-ETH`
`OHM-ETH`
`MIM-ETH`
`UST-ETH`
`FRAX-ETH`
`FEI-ETH`
`DOLA-ETH`
`DAI-ETH`
`USDC-ETH`
`USDT-ETH`
`ALUSD-ETH`

**Comment by 0x4007** [comment]
@hpiyankov pool2 purchase liquidity using UBQ from users (like OHM)

**Comment by hpiyankov** [comment]
Original setup:

We crate pool 1, which are a series of LP farms for popular pairs. Each of those pools gets 1000 uAR per day (for example) as a farming reward.
We crate pool 2, which is a uAR/ETH pair. We give away 100 000 uAR per day (for example) as farming rewards for this one.

The idea is: users farm the pools in pool 1 and then dump their rewards in pool 2 (+ETH).

New suggestions:

Pool 1: same as above
Pool 2: instead of giving the uAR as a reward for farming, we use it to purchase the liquidity outright. Example:

User A deposits 100 000 uAR + 1 ETH in the pool as liquidity. At current ETH prices, this would mean that he posted 8 000 USD in liquidity and that the price of 1 uAR is 0.04 USD. 

We offer the user to purchase his LP position in the pool at a 30% premium, so we give him 8000/0.04*1.3 = 260 000 uAR in exchange for purchasing his position. The catch.... he gets the uAR after 1 month (for example).

This is the equivalent of 2290% APY (if we calculate it as compounded APY).

**Reply by 0x4007** [reply]
@hpiyankov I'm rereading this now, and I don't understand what we benefit for purchasing back the liquidity. Is it to take the 1 ETH of that position? Do you think some type of bonding curve would be more profitable for us to grow the protocol controlled liquidity, vs your proposed buyback model?

**Reply by hpiyankov** [reply]
We have 2 general models on the market:

1. Give APY for liquidity (the standard model)
2. Purchase the liquidity at a premium (the OHM model)

What is the benefit of the OHM model? - In the end, you end up OWNIG the liquidity instead of just renting it.

When you say bonding curve, I do not understand what you mean by bonding curve in this context. Can you please elaborate a bit more?

**Reply by 0x4007** [reply]
Update from phone @hpiyankov call. Liquidity mining farming (traditional) incentive structure is mutually exclusive to the OHM buyback model. We will not do liquidity mining/farming incentives at all and instead will only offer rewards in the form of buyback. @hpiyankov @Zequez @marcomow @zapaz 

I can explain to you guys next time we all connect on our calls. 

@hpiyankov is currently setting final parameters to launch with. 

@zapaz do you think that creating setters and getters for these integers will be quick?

**Reply by 0x4007** [reply]
> When you say bonding curve, I do not understand what you mean by bonding curve in this context. Can you please elaborate a bit more?

Would rather keep this conversation simple and not go down that route, but I was thinking about how Fei literally raised over a billion dollars using their bonding curve when they launched. That to me seems like a pretty solid strategy to raise a bunch of cash.

**Reply by hpiyankov** [reply]
A bonding curve is a token sale strategy it is not a liquidity farming one.

**Reply by hpiyankov** [reply]
Regarding the parameters, for the "buyback model", there are two core parameters:
- Vesting period (with OHM this is 15 epochs or 5 days)
- Premium (with OHM this is based on the current treasury, which is not relevant in our case).

Essentially the APY of this approach is equal to:

APY = (1 + Premium)^(365/(Vesting/2))-1

Example: 5% premium and 6 days vesting would give us:

(1 + 0.05)^(365/(6/2)) - 1 = 37 746% annual APY

Higher premium = more APY
Higher vesting = less APY

We can manipulate the above to our advantage, given that the pool will be fixed duration. The core question is: do we think people would manage their position very actively (like daily/hourly):

IF yes, then I would do 30% premium with 30 days vesting, which would net 59 140% APY
IF no, then I would do 2.5% premium with 3 days vesting, which would net 40 588% APY

**Comment by hpiyankov** [comment]
@pavlovcik 4.5% premium and 5 days vesting gives you 61698.57% APY
